src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -2566,6 +2648,52 @@ internal static void SetPrimitiveValueConverter(this IEdmModel model, string ful
             mapForModel[fullTypeName] = converter;
         }
 
+        /// <summary>
+        /// Gets the declared alternate keys of the most defined entity with a declared key present.
+        /// </summary>
+        /// <param name="type">Reference to the calling object.</param>
+        /// <param name="model">The model to be used.</param>
+        /// <returns>Key of this type.</returns>
+        private static IEnumerable<IDictionary<string, IEdmProperty>> GetDeclaredAlternateKeysForType(IEdmEntityType type, IEdmModel model)
+        {
+            IEdmValueAnnotation annotationValue = model.FindVocabularyAnnotations<IEdmValueAnnotation>(type, AlternateKeysVocabularyModel.AlternateKeysTerm).FirstOrDefault();
+
+            if (annotationValue != null)
+            {
+                List<IDictionary<string, IEdmProperty>> declaredAlternateKeys = new List<IDictionary<string, IEdmProperty>>();
+
+                IEdmCollectionExpression keys = annotationValue.Value as IEdmCollectionExpression;
+                foreach (IEdmRecordExpression key in keys.Elements)
+                {
+                    Debug.Assert(key.Properties.FirstOrDefault() != null, "expected non null alternatekey");

```

Comment:
can change to key.Properties.FirstOrDefault(e => e.Name == AlternateKeysVocabularyConstants.AlternateKeyTypeKeyPropertyName)